index.bs

```diff
@@ -5483,6 +5504,379 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>DELETE</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |authenticatorId| does not match any [=Virtual Authenticator=] stored in the [=Virtual Authenticator
+     Database=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Remove the [=Virtual Authenticator=] identified by |authenticatorId| from the [=Virtual Authenticator Database=]
+ 1. Return [=success=].
+
+## <dfn>Add Credential</dfn> ## {#sctn-automation-add-credential}
+
+The [=Add Credential=] WebDriver [=extension command=] injects a [=Credential Key Pair=] and associated [=Authenticator Data=]
+into an existing [=Virtual Authenticator=]. It is defined as follows:
+
+<figure id="table-addCredential" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>HTTP Method</th>
+                <th>URI Template</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>POST</td>
+                <td>`/session/{session id}/webauthn/authenticator/{authenticatorId}/credential`</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The <dfn>Credential Parameters</dfn> is a JSON [=Object=] passed to the [=remote end steps=] as |parameters|. It contains the following |key| and |value| pairs:
+
+<figure id="table-credentialParameters" class="table">
+    <table class="data">
+        <thead>
+            <tr>
+                <th>Key</th>
+                <th>Value Type</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td>|credentialId|</td>
+                <td>string</td>
+            </tr>
+            <tr>
+                <td>|rpId|</td>
+                <td>string</td>
+            </tr>
+            <tr>
+                <td>|privateKey|</td>
+                <td>string</td>
+            </tr>
+            <tr>
+                <td>|signCount|</td>
+                <td>number</td>
+            </tr>
+        </tbody>
+    </table>
+</figure>
+
+The [=remote end steps=] are:
+
+ 1. If |parameters| is not a JSON [=Object=], return a [=WebDriver error=] with [=WebDriver error code=]
+     [=invalid argument=].
+     Note: |parameters| is a [=Credential Parameters=] object.
+ 1. Let |credentialId| be the result of decoding [=Base64url Encoding=] on the |parameters|' |credentialId| property.
+ 1. If |credentialId| is failure, return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
+ 1. Let |rpId| be the the |parameters|' |rpId| property.
+ 1. If |rpId| is not a valid [=RP ID=], return a [=WebDriver error=] with [=WebDriver error code=] [=invalid argument=].
```

Comment:
Hm, I spose havin the [=RP ID=] link gives the reader enough context in order to figure out how to validate an alleged RP ID value?

Reply:
That is my assumption, yes :)